### PR TITLE
Make some objects immutable

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: src/Report/ReportSummary.php [hunk 1]
```diff
@@ -40,21 +40,37 @@
...
-    public static function create()
-    {
-        return new self();
+     * @param array $changed
+     * @param int   $time
+     * @param int   $memory
+     * @param bool  $addAppliedFixers
+     * @param bool  $isDryRun
+     * @param bool  $isDecoratedOutput
+     */
```
Comment (SpacePossum) [inline src/Report/ReportSummary.php]: these could you a bit of explaining, like is time processing time, UNIX timestamp, is it in seconds or minisecs., same for memory; is it in bytes or bits?

File: src/Report/TextReporter.php [hunk 2]
```diff
@@ -115,7 +115,7 @@ function ($string) {
      */
     private function getFooter($time, $memory, $isDryRun)
     {
-        if ($time === null || $memory === null) {
+        if ($time === 0 || $memory === 0) {
```
Comment (SpacePossum) [inline src/Report/TextReporter.php]: Yoda :)

File: src/Report/XmlReporter.php [hunk 3]
```diff
@@ -58,11 +58,11 @@ public function generate(ReportSummary $reportSummary)
             }
         }
 
-        if (null !== $reportSummary->getTime()) {
+        if (0 !== $reportSummary->getTime()) {
```
Comment (SpacePossum) [inline src/Report/XmlReporter.php]: Yoda :)

**Comment by SpacePossum** [comment]
Looks good, just a few minors :)

File: src/Report/XmlReporter.php (hunk 3 above)
Comment (keradus) [inline src/Report/XmlReporter.php]: it is yoda ;)

Comment (SpacePossum) [inline src/Report/XmlReporter.php]: haha sorry man, must've lost my brain again ;)

**Comment by SpacePossum** [comment]
:+1: looking good :)

**Pull request closed without merging**